### ISSUE TITLE
1.8.0-rc3: System error: mkdir /sys/fs/docker: operation not permitted

### ISSUE THREAD

**Issue by ibuildthecloud** [docker/docker#15433]
I can't get Docker 1.8.0-rc3 (no 1.8.x has worked) to work in RancherOS where Docker 1.7.1 works fine.  There's probably some odd ball thing different about the environment of RancherOS that seems to mess things up.  The system-docker, running as PID 1 works just fine.  It is the user docker that runs in a Docker-in-Docker context that fails.  I just get the following error whenever I try to start a container:

   Error response from daemon: Cannot start container 4d5fd0f969024e5b288948fd6512a76e597b994c91c7368a6c7b987c0d6037ad: [8] System error: mkdir /sys/fs/docker: operation not permitted

My guess here is that for some reason Docker/libcontainer can't figure out where the cgroup fs root is.  Here's the mountinfo from a running docker daemon:

```
169 83 0:47 / / rw,relatime - overlay overlay rw,lowerdir=/var/lib/system-docker/overlay/264bcf699b1942d0d5ee2caed60b491c02bbe498e9bac137826343130a411296/root,upperdir=/var/lib/system-docker/overlay/c844461ecea9664c95a5d3f4fc02264f913f1be349fa3a2319f5fc60605b93ab/upper,workdir=/var/lib/system-docker/overlay/c844461ecea9664c95a5d3f4fc02264f913f1be349fa3a2319f5fc60605b93ab/work
170 169 0:5 / /proc rw,nosuid,nodev,noexec,relatime - proc proc rw
171 169 0:48 / /dev rw,nosuid - tmpfs tmpfs rw,mode=755
172 171 0:49 / /dev/pts rw,nosuid,noexec,relatime - devpts devpts rw,gid=5,mode=620,ptmxmode=666
173 171 0:50 / /dev/shm rw,nosuid,nodev,noexec,relatime - tmpfs shm rw,size=65536k
174 171 0:14 / /dev/mqueue rw,nosuid,nodev,noexec,relatime - mqueue mqueue rw
175 169 0:16 / /sys rw,nosuid,nodev,noexec,relatime - sysfs sysfs rw
176 175 0:51 / /sys/fs/cgroup rw,nosuid,nodev,noexec,relatime - tmpfs tmpfs rw
177 176 0:18 /docker/c844461ecea9664c95a5d3f4fc02264f913f1be349fa3a2319f5fc60605b93ab /sys/fs/cgroup/freezer rw,relatime - cgroup none rw,freezer
178 176 0:19 /docker/c844461ecea9664c95a5d3f4fc02264f913f1be349fa3a2319f5fc60605b93ab /sys/fs/cgroup/net_cls,net_prio rw,relatime - cgroup none rw,net_cls,net_prio
179 176 0:20 /docker/c844461ecea9664c95a5d3f4fc02264f913f1be349fa3a2319f5fc60605b93ab /sys/fs/cgroup/blkio rw,relatime - cgroup none rw,blkio
180 176 0:21 /docker/c844461ecea9664c95a5d3f4fc02264f913f1be349fa3a2319f5fc60605b93ab /sys/fs/cgroup/perf_event rw,relatime - cgroup none rw,perf_event
181 176 0:22 /docker/c844461ecea9664c95a5d3f4fc02264f913f1be349fa3a2319f5fc60605b93ab /sys/fs/cgroup/cpuset rw,relatime - cgroup none rw,cpuset
182 176 0:23 /docker/c844461ecea9664c95a5d3f4fc02264f913f1be349fa3a2319f5fc60605b93ab /sys/fs/cgroup/cpu,cpuacct rw,relatime - cgroup none rw,cpu,cpuacct
183 176 0:24 /docker/c844461ecea9664c95a5d3f4fc02264f913f1be349fa3a2319f5fc60605b93ab /sys/fs/cgroup/memory rw,relatime - cgroup none rw,memory
184 176 0:25 /docker/c844461ecea9664c95a5d3f4fc02264f913f1be349fa3a2319f5fc60605b93ab /sys/fs/cgroup/devices rw,relatime - cgroup none rw,devices
185 169 254:0 /opt /opt rw,relatime - ext4 /dev/vda rw,data=ordered
186 169 254:0 /home /home rw,relatime - ext4 /dev/vda rw,data=ordered
187 169 254:0 /usr-v0.4.0-dev/bin/ros /sbin/reboot ro,relatime - ext4 /dev/vda rw,data=ordered
188 169 254:0 /usr-v0.4.0-dev/bin/ros /sbin/halt ro,relatime - ext4 /dev/vda rw,data=ordered
189 169 254:0 /usr-v0.4.0-dev/bin/ros /sbin/poweroff ro,relatime - ext4 /dev/vda rw,data=ordered
190 169 254:0 /usr-v0.4.0-dev/lib/modules /lib/modules rw,relatime - ext4 /dev/vda rw,data=ordered
191 169 0:15 / /var/run rw,relatime - tmpfs none rw
192 191 0:4 / /var/run/docker/netns/default rw - nsfs nsfs rw
193 169 254:0 /usr-v0.4.0-dev/lib/firmware /lib/firmware rw,relatime - ext4 /dev/vda rw,data=ordered
194 169 254:0 /usr-v0.4.0-dev/bin/ros /sbin/shutdown ro,relatime - ext4 /dev/vda rw,data=ordered
195 169 254:0 /var/log /var/log rw,relatime - ext4 /dev/vda rw,data=ordered
196 169 254:0 /usr-v0.4.0-dev/bin/ros /sbin/netconf ro,relatime - ext4 /dev/vda rw,data=ordered
197 169 0:6 / /host/dev rw,relatime - devtmpfs devtmpfs rw,size=455348k,nr_inodes=113837,mode=755
198 197 0:12 / /host/dev/pts rw,relatime - devpts none rw,mode=600,ptmxmode=000
199 169 254:0 /var/lib/rancher /var/lib/rancher rw,relatime - ext4 /dev/vda rw,data=ordered
200 169 254:0 /usr-v0.4.0-dev/bin/ros /usr/bin/ros ro,relatime - ext4 /dev/vda rw,data=ordered
201 169 254:0 /usr-v0.4.0-dev/bin/ros /usr/bin/dockerlaunch ro,relatime - ext4 /dev/vda rw,data=ordered
202 169 254:0 /var/lib/docker /var/lib/docker rw,relatime - ext4 /dev/vda rw,data=ordered
203 169 254:0 /usr-v0.4.0-dev/bin/docker /usr/bin/docker.dist ro,relatime - ext4 /dev/vda rw,data=ordered
204 169 254:0 /usr-v0.4.0-dev/bin/ros /usr/bin/rancherctl ro,relatime - ext4 /dev/vda rw,data=ordered
205 169 254:0 /usr-v0.4.0-dev/bin/ros /usr/bin/system-docker ro,relatime - ext4 /dev/vda rw,data=ordered
206 169 254:0 /usr-v0.4.0-dev/bin/ros /usr/sbin/wait-for-docker ro,relatime - ext4 /dev/vda rw,data=ordered
207 169 254:0 /usr-v0.4.0-dev/bin/ros /usr/bin/cloud-init ro,relatime - ext4 /dev/vda rw,data=ordered
208 169 254:0 /usr-v0.4.0-dev/bin/ros /usr/bin/respawn ro,relatime - ext4 /dev/vda rw,data=ordered
209 199 254:0 /var/lib/rancher/conf /var/lib/rancher/conf rw,relatime - ext4 /dev/vda rw,data=ordered
210 169 254:0 /usr-v0.4.0-dev/share/ros/os-config.yml /usr/share/ros/os-config.yml rw,relatime - ext4 /dev/vda rw,data=ordered
211 169 254:0 /etc/ssl/certs/ca-certificates.crt /etc/ssl/certs/ca-certificates.crt.rancher rw,relatime - ext4 /dev/vda rw,data=ordered
212 169 254:0 /var/lib/system-docker/containers/c844461ecea9664c95a5d3f4fc02264f913f1be349fa3a2319f5fc60605b93ab/resolv.conf /etc/resolv.conf rw,relatime - ext4 /dev/vda rw,data=ordered
213 169 254:0 /var/lib/system-docker/containers/c844461ecea9664c95a5d3f4fc02264f913f1be349fa3a2319f5fc60605b93ab/hostname /etc/hostname rw,relatime - ext4 /dev/vda rw,data=ordered
214 169 254:0 /var/lib/system-docker/containers/c844461ecea9664c95a5d3f4fc02264f913f1be349fa3a2319f5fc60605b93ab/hosts /etc/hosts rw,relatime - ext4 /dev/vda rw,data=ordered
79 171 0:6 / /dev rw,relatime - devtmpfs devtmpfs rw,size=455348k,nr_inodes=113837,mode=755
84 79 0:12 / /dev/pts rw,relatime - devpts none rw,mode=600,ptmxmode=000
217 169 0:52 / /run rw,relatime - tmpfs none rw
```

and `/proc/mounts` if that helps

```
overlay / overlay rw,relatime,lowerdir=/var/lib/system-docker/overlay/264bcf699b1942d0d5ee2caed60b491c02bbe498e9bac137826343130a411296/root,upperdir=/var/lib/system-docker/overlay/c844461ecea9664c95a5d3f4fc02264f913f1be349fa3a2319f5fc60605b93ab/upper,workdir=/var/lib/system-docker/overlay/c844461ecea9664c95a5d3f4fc02264f913f1be349fa3a2319f5fc60605b93ab/work 0 0
proc /proc proc rw,nosuid,nodev,noexec,relatime 0 0
tmpfs /dev tmpfs rw,nosuid,mode=755 0 0
devpts /dev/pts devpts rw,nosuid,noexec,relatime,gid=5,mode=620,ptmxmode=666 0 0
shm /dev/shm tmpfs rw,nosuid,nodev,noexec,relatime,size=65536k 0 0
mqueue /dev/mqueue mqueue rw,nosuid,nodev,noexec,relatime 0 0
sysfs /sys sysfs rw,nosuid,nodev,noexec,relatime 0 0
tmpfs /sys/fs/cgroup tmpfs rw,nosuid,nodev,noexec,relatime 0 0
none /sys/fs/cgroup/freezer cgroup rw,relatime,freezer 0 0
none /sys/fs/cgroup/net_cls,net_prio cgroup rw,relatime,net_cls,net_prio 0 0
none /sys/fs/cgroup/blkio cgroup rw,relatime,blkio 0 0
none /sys/fs/cgroup/perf_event cgroup rw,relatime,perf_event 0 0
none /sys/fs/cgroup/cpuset cgroup rw,relatime,cpuset 0 0
none /sys/fs/cgroup/cpu,cpuacct cgroup rw,relatime,cpu,cpuacct 0 0
none /sys/fs/cgroup/memory cgroup rw,relatime,memory 0 0
none /sys/fs/cgroup/devices cgroup rw,relatime,devices 0 0
/dev/vda /opt ext4 rw,relatime,data=ordered 0 0
/dev/vda /home ext4 rw,relatime,data=ordered 0 0
/dev/vda /sbin/reboot ext4 ro,relatime,data=ordered 0 0
/dev/vda /sbin/halt ext4 ro,relatime,data=ordered 0 0
/dev/vda /sbin/poweroff ext4 ro,relatime,data=ordered 0 0
/dev/vda /lib/modules ext4 rw,relatime,data=ordered 0 0
none /var/run tmpfs rw,relatime 0 0
nsfs /var/run/docker/netns/default nsfs rw 0 0
/dev/vda /lib/firmware ext4 rw,relatime,data=ordered 0 0
/dev/vda /sbin/shutdown ext4 ro,relatime,data=ordered 0 0
/dev/vda /var/log ext4 rw,relatime,data=ordered 0 0
/dev/vda /sbin/netconf ext4 ro,relatime,data=ordered 0 0
devtmpfs /host/dev devtmpfs rw,relatime,size=455348k,nr_inodes=113837,mode=755 0 0
none /host/dev/pts devpts rw,relatime,mode=600,ptmxmode=000 0 0
/dev/vda /var/lib/rancher ext4 rw,relatime,data=ordered 0 0
/dev/vda /usr/bin/ros ext4 ro,relatime,data=ordered 0 0
/dev/vda /usr/bin/dockerlaunch ext4 ro,relatime,data=ordered 0 0
/dev/vda /var/lib/docker ext4 rw,relatime,data=ordered 0 0
/dev/vda /usr/bin/docker.dist ext4 ro,relatime,data=ordered 0 0
/dev/vda /usr/bin/rancherctl ext4 ro,relatime,data=ordered 0 0
/dev/vda /usr/bin/system-docker ext4 ro,relatime,data=ordered 0 0
/dev/vda /usr/sbin/wait-for-docker ext4 ro,relatime,data=ordered 0 0
/dev/vda /usr/bin/cloud-init ext4 ro,relatime,data=ordered 0 0
/dev/vda /usr/bin/respawn ext4 ro,relatime,data=ordered 0 0
/dev/vda /var/lib/rancher/conf ext4 rw,relatime,data=ordered 0 0
/dev/vda /usr/share/ros/os-config.yml ext4 rw,relatime,data=ordered 0 0
/dev/vda /etc/ssl/certs/ca-certificates.crt.rancher ext4 rw,relatime,data=ordered 0 0
/dev/vda /etc/resolv.conf ext4 rw,relatime,data=ordered 0 0
/dev/vda /etc/hostname ext4 rw,relatime,data=ordered 0 0
/dev/vda /etc/hosts ext4 rw,relatime,data=ordered 0 0
devtmpfs /dev devtmpfs rw,relatime,size=455348k,nr_inodes=113837,mode=755 0 0
none /dev/pts devpts rw,relatime,mode=600,ptmxmode=000 0 0
none /run tmpfs rw,relatime 0 0
```

I will dig into this tomorrow more in depth and start debugging the libcontainer code.

ping @LK4D4 @crosbymichael

**Comment by ibuildthecloud** [comment]
@GordonTheTurtle I shall counter your #ENEEDMOREINFO with my #EYOURRULESDONTAPPLYTOME

**Comment by thaJeztah** [comment]
@ibuildthecloud :smile: feel free [to do so](https://github.com/docker/leeroy/blob/d5ee4f3bf05276a9891b1eca949a5889b66aec19/github/issue.go#L10)

**Comment by thaJeztah** [comment]
ping @calavera as release captain (I added it to the milestone so it doesn't get out of sight)

(but, knowing Darren, I don't think getting out of sight will be a problem :wink:)

**Comment by calavera** [comment]
Hey @ibuildthecloud, I have a "random" question for you, does RancherOS use apparmor?

**Comment by calavera** [comment]
Because if it does, and you installed one of the previous prereleases, you might have a broken profile installed that you should remove. :pensive:

**Comment by ibuildthecloud** [comment]
@calavera No apparmor.  "Coming Soon"

**Comment by tiborvass** [comment]
I could reproduce with rancheros

**Comment by thaJeztah** [comment]
![image](https://cloud.githubusercontent.com/assets/1804568/9175242/e424d72e-3f84-11e5-9fd8-d36c2d80dac9.png)

**Comment by LK4D4** [comment]
@ibuildthecloud Could you provide `/proc/self/mountinfo` from inside your "system" container?
Am I understanding right, that second(not third) nested docker fails to start any container?

**Comment by tiborvass** [comment]
@ibuildthecloud I found the PR that broke it: #14716

@LK4D4 I did `cat /proc/self/mountinfo` after a `./scripts/ssh` (@ibuildthecloud let me know if that's correct)

```
232 229 0:20 /docker/7805ced5eb4bdd2d03640423f55409c82397a9714cdc6267a0495dff80ae00f5 /var/lib/system-docker/overlay/7805ced5eb4bdd2d03640423f55409c82397a9714cdc6267a0495dff80ae00f5/merged/sys/fs/cgroup/freezer rw,relatime - cgroup none rw,freezer
233 229 0:21 /docker/7805ced5eb4bdd2d03640423f55409c82397a9714cdc6267a0495dff80ae00f5 /var/lib/system-docker/overlay/7805ced5eb4bdd2d03640423f55409c82397a9714cdc6267a0495dff80ae00f5/merged/sys/fs/cgroup/net_cls,net_prio rw,relatime - cgroup none rw,net_cls,net_prio
234 229 0:22 /docker/7805ced5eb4bdd2d03640423f55409c82397a9714cdc6267a0495dff80ae00f5 /var/lib/system-docker/overlay/7805ced5eb4bdd2d03640423f55409c82397a9714cdc6267a0495dff80ae00f5/merged/sys/fs/cgroup/blkio rw,relatime - cgroup none rw,blkio
235 229 0:23 /docker/7805ced5eb4bdd2d03640423f55409c82397a9714cdc6267a0495dff80ae00f5 /var/lib/system-docker/overlay/7805ced5eb4bdd2d03640423f55409c82397a9714cdc6267a0495dff80ae00f5/merged/sys/fs/cgroup/perf_event rw,relatime - cgroup none rw,perf_event
236 229 0:24 /docker/7805ced5eb4bdd2d03640423f55409c82397a9714cdc6267a0495dff80ae00f5 /var/lib/system-docker/overlay/7805ced5eb4bdd2d03640423f55409c82397a9714cdc6267a0495dff80ae00f5/merged/sys/fs/cgroup/cpuset rw,relatime - cgroup none rw,cpuset
237 229 0:25 /docker/7805ced5eb4bdd2d03640423f55409c82397a9714cdc6267a0495dff80ae00f5 /var/lib/system-docker/overlay/7805ced5eb4bdd2d03640423f55409c82397a9714cdc6267a0495dff80ae00f5/merged/sys/fs/cgroup/cpu,cpuacct rw,relatime - cgroup none rw,cpu,cpuacct
238 190 254:0 /home /var/lib/system-docker/overlay/7805ced5eb4bdd2d03640423f55409c82397a9714cdc6267a0495dff80ae00f5/merged/home rw,relatime - ext4 /dev/vda rw,data=ordered
239 190 254:0 /opt /var/lib/system-docker/overlay/7805ced5eb4bdd2d03640423f55409c82397a9714cdc6267a0495dff80ae00f5/merged/opt rw,relatime - ext4 /dev/vda rw,data=ordered
240 190 254:0 /usr-v0.4.0-dev/bin/ros /var/lib/system-docker/overlay/7805ced5eb4bdd2d03640423f55409c82397a9714cdc6267a0495dff80ae00f5/merged/sbin/netconf ro,relatime - ext4 /dev/vda rw,data=ordered
241 190 254:0 /usr-v0.4.0-dev/bin/ros /var/lib/system-docker/overlay/7805ced5eb4bdd2d03640423f55409c82397a9714cdc6267a0495dff80ae00f5/merged/sbin/reboot ro,relatime - ext4 /dev/vda rw,data=ordered
242 190 254:0 /usr-v0.4.0-dev/bin/ros /var/lib/system-docker/overlay/7805ced5eb4bdd2d03640423f55409c82397a9714cdc6267a0495dff80ae00f5/merged/sbin/shutdown ro,relatime - ext4 /dev/vda rw,data=ordered
243 190 254:0 /var/log /var/lib/system-docker/overlay/7805ced5eb4bdd2d03640423f55409c82397a9714cdc6267a0495dff80ae00f5/merged/var/log rw,relatime - ext4 /dev/vda rw,data=ordered
244 190 254:0 /usr-v0.4.0-dev/bin/ros /var/lib/system-docker/overlay/7805ced5eb4bdd2d03640423f55409c82397a9714cdc6267a0495dff80ae00f5/merged/sbin/halt ro,relatime - ext4 /dev/vda rw,data=ordered
245 190 254:0 /usr-v0.4.0-dev/bin/ros /var/lib/system-docker/overlay/7805ced5eb4bdd2d03640423f55409c82397a9714cdc6267a0495dff80ae00f5/merged/sbin/poweroff ro,relatime - ext4 /dev/vda rw,data=ordered
246 190 254:0 /usr-v0.4.0-dev/lib/firmware /var/lib/system-docker/overlay/7805ced5eb4bdd2d03640423f55409c82397a9714cdc6267a0495dff80ae00f5/merged/lib/firmware rw,relatime - ext4 /dev/vda rw,data=ordered
247 190 0:6 / /var/lib/system-docker/overlay/7805ced5eb4bdd2d03640423f55409c82397a9714cdc6267a0495dff80ae00f5/merged/host/dev rw,relatime - devtmpfs devtmpfs rw,size=1487464k,nr_inodes=371866,mode=755
248 247 0:12 / /var/lib/system-docker/overlay/7805ced5eb4bdd2d03640423f55409c82397a9714cdc6267a0495dff80ae00f5/merged/host/dev/pts rw,relatime - devpts none rw,mode=600,ptmxmode=000
249 190 0:15 / /var/lib/system-docker/overlay/7805ced5eb4bdd2d03640423f55409c82397a9714cdc6267a0495dff80ae00f5/merged/var/run rw,relatime - tmpfs none rw
250 249 0:4 / /var/lib/system-docker/overlay/7805ced5eb4bdd2d03640423f55409c82397a9714cdc6267a0495dff80ae00f5/merged/var/run/docker/netns/default rw - nsfs nsfs rw
251 190 254:0 /usr-v0.4.0-dev/lib/modules /var/lib/system-docker/overlay/7805ced5eb4bdd2d03640423f55409c82397a9714cdc6267a0495dff80ae00f5/merged/lib/modules rw,relatime - ext4 /dev/vda rw,data=ordered
252 185 0:46 / /var/lib/system-docker/overlay/f774a899f8cdb7cad32a58fc7f3d6032353d125c60ac5b23e2010f6dd994a780/merged rw,relatime - overlay overlay rw,lowerdir=/var/lib/system-docker/overlay/fa9bddd427ed3807f44f483b40b44517cd3abb98e19e1e71e8d4e0e6c657ea01/root,upperdir=/var/lib/system-docker/overlay/f774a899f8cdb7cad32a58fc7f3d6032353d125c60ac5b23e2010f6dd994a780/upper,workdir=/var/lib/system-docker/overlay/f774a899f8cdb7cad32a58fc7f3d6032353d125c60ac5b23e2010f6dd994a780/work
253 112 254:0 /usr-v0.4.0-dev/bin/ros /usr/bin/respawn ro,relatime - ext4 /dev/vda rw,data=ordered
254 112 254:0 /usr-v0.4.0-dev/bin/ros /usr/sbin/wait-for-docker ro,relatime - ext4 /dev/vda rw,data=ordered
255 112 254:0 /usr-v0.4.0-dev/bin/ros /usr/bin/cloud-init ro,relatime - ext4 /dev/vda rw,data=ordered
256 112 254:0 /usr-v0.4.0-dev/bin/ros /usr/bin/rancherctl ro,relatime - ext4 /dev/vda rw,data=ordered
257 112 254:0 /usr-v0.4.0-dev/bin/ros /usr/bin/ros ro,relatime - ext4 /dev/vda rw,data=ordered
258 112 254:0 /usr-v0.4.0-dev/bin/docker /usr/bin/docker ro,relatime - ext4 /dev/vda rw,data=ordered
259 112 254:0 /usr-v0.4.0-dev/bin/ros /usr/bin/system-docker ro,relatime - ext4 /dev/vda rw,data=ordered
260 112 254:0 /var/lib/docker /var/lib/docker rw,relatime - ext4 /dev/vda rw,data=ordered
261 112 254:0 /var/lib/rancher /var/lib/rancher rw,relatime - ext4 /dev/vda rw,data=ordered
262 112 254:0 /usr-v0.4.0-dev/bin/ros /usr/bin/dockerlaunch ro,relatime - ext4 /dev/vda rw,data=ordered
263 112 254:0 /usr-v0.4.0-dev/share/ros/os-config.yml /usr/share/ros/os-config.yml rw,relatime - ext4 /dev/vda rw,data=ordered
264 261 254:0 /var/lib/rancher/conf /var/lib/rancher/conf rw,relatime - ext4 /dev/vda rw,data=ordered
265 112 254:0 /etc/ssl/certs/ca-certificates.crt /etc/ssl/certs/ca-certificates.crt.rancher rw,relatime - ext4 /dev/vda rw,data=ordered
266 112 254:0 /var/lib/system-docker/containers/7805ced5eb4bdd2d03640423f55409c82397a9714cdc6267a0495dff80ae00f5/resolv.conf /etc/resolv.conf rw,relatime - ext4 /dev/vda rw,data=ordered
267 112 254:0 /var/lib/system-docker/containers/7805ced5eb4bdd2d03640423f55409c82397a9714cdc6267a0495dff80ae00f5/hostname /etc/hostname rw,relatime - ext4 /dev/vda rw,data=ordered
268 112 254:0 /var/lib/system-docker/containers/7805ced5eb4bdd2d03640423f55409c82397a9714cdc6267a0495dff80ae00f5/hosts /etc/hosts rw,relatime - ext4 /dev/vda rw,data=ordered
48 114 0:6 / /dev rw,relatime - devtmpfs devtmpfs rw,size=1487464k,nr_inodes=371866,mode=755
49 48 0:12 / /dev/pts rw,relatime - devpts none rw,mode=600,ptmxmode=000
50 112 0:45 / /run rw,relatime - tmpfs none rw
51 260 0:56 / /var/lib/docker/overlay/ad2e6eba1fbd212eb7f1f9f29f615019aa88bf45ae4d8c187df972509043e70f/merged rw,relatime - overlay overlay rw,lowerdir=/var/lib/docker/overlay/8c2e06607696bd4afb3d03b687e361cc43cf8ec1a4a725bc96e39f05ba97dd55/root,upperdir=/var/lib/docker/overlay/ad2e6eba1fbd212eb7f1f9f29f615019aa88bf45ae4d8c187df972509043e70f/upper,workdir=/var/lib/docker/overlay/ad2e6eba1fbd212eb7f1f9f29f615019aa88bf45ae4d8c187df972509043e70f/work
89 182 0:4 / /var/run/docker/netns/ad2e6eba1fbd rw - nsfs nsfs rw
```

**Comment by LK4D4** [comment]
@ibuildthecloud Do rancheros mount cgroups itself inside system container? How it doing this? could you provide script?

**Comment by ibuildthecloud** [comment]
@LK4D4 The code is here: https://github.com/rancher/docker-from-scratch/blob/master/scratch.go#L72

**Comment by LK4D4** [comment]
I did fix in opencontainers/runc#191

**Comment by icecrime** [comment]
Not yet fixed in docker master (need to bump runC), but will be fixed in 1.8.0.

**Comment by icecrime** [comment]
Change of plan: this is _not_ fixed in 1.8.0... Sorry, we have a regression with the patch.

**Comment by icecrime** [comment]
Closed by opencontainers/runc#194 (tagged as `0.0.2.1`), bumped into 1.8.2.
